### PR TITLE
Make all emails with the same pattern on subject

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
Emails sent by the app/email.go didnt follow the same pattern in the subject line, now it follow :)

#### Ticket Link
N/A

#### Checklist
- [X] Added or updated unit tests (required for all new features)

### CODE REVIEW

**Comment by jasonblais** [comment]
Thanks @cpanato! Is this just adding the SiteName to all the email subjects?

**Comment by cpanato** [comment]
@jasonblais  yes add the format " [ SiteName]  <type of messge>"

**Comment by jasonblais** [comment]
Sounds good! @doosp will help with first round of testing -- if you have any questions about test cases, please touch base with @esethna who's more familiar with email notifications

**Comment by doosp** [comment]
Hi, @cpanato 

Good to see you again. :)
Thanks for your help with consistent email subject format.

All the email showed correct [SiteName] in the title except one:

image

Repro step:
1. Go to "I forgot my password" on the login page
2. Enter an email address and click "Rest my password"

Would you please help looking into this?

Thank you.

**Comment by cpanato** [comment]
@jasonblais @doosp the reset password is not in the app/email.go file it is in the user.go, but I think it makes sense to move to the email file because all the stuffs related to sending email are there.

If you guys agree I can move.

**Comment by jasonblais** [comment]
@jwilander can you confirm whether it would be okay to move reset password email stuff over to app/email.go, or do we want to keep it in user.go?

**Comment by jwilander** [comment]
Yes, it's OK to move the building of the subject/body and the actual sending of the email into `app/email.go`. Any generation of hashes or any actual logic on passwords should be left in `app/user.go`

**Comment by doosp** [comment]
@cpanato 

Thanks for the update. 

I tested the reset password and got [SiteName] at the beginning of email subject

image

Also tested other email notifications and they all have the same format. Awesome work! 👍 

@jasonblais Would you please review? Thanks.

**Comment by esethna** [comment]
Hey @cpanato, thanks for the PR! This is working great. I just ran into an issue in the system console. I tested on our CI servers and was not able to repro, so it might be related to your PR? Notice that the main menu is moved to the left overlapping the text:

image

**Comment by cpanato** [comment]
@esethna using the latest spinserver I did not see the issue you mentioned

system console mattermost

**Comment by esethna** [comment]
You are right, thanks for checking! There must've been as issue on master which was resolved.

File: api/user.go [hunk 1]
```diff
@@ -814,7 +817,22 @@ func sendPasswordReset(c *Context, w http.ResponseWriter, r *http.Request) {
...
+		return
+	}
+
+	var recovery *model.PasswordRecovery
+	if recovery, err = app.CreatePasswordRecovery(user.Id); err != nil {
+		c.Err = err
+		return
+	}
+
+	if sent, err := app.SendPasswordReset(email, recovery, user.Locale, c.GetSiteURL()); err != nil {
```
Comment (jwilander) [inline api/user.go]: @cpanato Can you move all the above code back into the `app` package? You can still have a separate function for sending the email in `app/email.go` but that can be called `SendPasswordResetEmail()` and used from within `app/user.go` where this code originally was

**Comment by jwilander** [comment]
Thanks @cpanato !

File: app/email.go [hunk 2]
```diff
@@ -101,8 +101,8 @@ func SendVerifyEmail(userId, userEmail, locale, siteURL string) *model.AppError
 func SendSignInChangeEmail(email, method, locale, siteURL string) *model.AppError {
 	T := utils.GetUserTranslations(locale)
 
-	subject := T("api.templates.singin_change_email.subject",
-		map[string]interface{}{"SiteName": utils.ClientCfg["SiteName"]})
+	subject := fmt.Sprintf("[%v] %v", utils.Cfg.TeamSettings.SiteName, T("api.templates.singin_change_email.subject",
```
Comment (coreyhulen) [inline app/email.go]: I might be missing something but why isn't "[%v] %v" part of the localization string vs doing a `fmt.Sprintf`?

Maybe @enahum can comment?

Comment (enahum) [inline app/email.go]: So the string is `You updated your sign-in method on {{ .SiteName }}` right?

I guess that's fine to leave it this way but it feels weird having the same variable being used twice

Comment (coreyhulen) [inline app/email.go]: String formatting should be done in the loc stuff?  No @enahum 

like

```go
api.templates.singin_change_email.subject := "[{{ .SiteName }}] You updated your sign-in method on {{ .SiteName }}"

subject := T("api.templates.singin_change_email.subject", map[string]interface{}{"SiteName": utils.ClientCfg["SiteName"]}))
```

What happens when we want to change `[]` to `()` or `[]` has a special meaning in a different language?

**Comment by cpanato** [comment]
@coreyhulen What is the changes you want to apply?

**Comment by coreyhulen** [comment]
I think all the `[%v] %v"` should be part of the loc string and not using the Sprintf to inject it.

**Comment by coreyhulen** [comment]
Lets make all the [%v] %v" part of the format string and not use Sprintf to format the string.

**Comment by jasonblais** [comment]
@lindalu and I will help review emails and remove references to sitename and teamname from the subject line when they're not necessary (since we're adding `[Site_Name]` to the start of the subject lines.

**Comment by jasonblais** [comment]
@cpanato Sorry, can you help rebase your branch to resolve merge conflicts? We'll then generate a new test server for it.

**Comment by cpanato** [comment]
@coreyhulen update the subjects per your review

**Comment by jasonblais** [comment]
Thanks @cpanato! 

@lindalumitchell: not urgent, when you get the chance:

Can you help review the different emails and prepare a list where either team names (e.g. `Contributors`) or the site name (e.g. `Mattermost`) is no longer necessary on the subject line, since we're adding [SITE_NAME] to the start of each subject line now

Example: Email for resetting your password reads `[Mattermost] You updated your password for Contributors on Mattermost` where both the team name and site name are unnecessary. Subject line `[Mattermost] You updated your password` would be sufficient.

**Comment by lindalumitchell** [comment]
@jasonblais Here's the list of email subjects that I think could be changed per the discussion:

1. [Mattermost] Your username has changed for Mattermost: 
  - Change to `[Mattermost] Your username has changed`
2. [Mattermost] Your email address has changed for Mattermost:
  - Change to `[Mattermost] Your email address has changed`
3. [Mattermost] Verify new email address for Mattermost:
  - Change to `[Mattermost] Verify new email address`
4. [Mattermost] Your password has been updated for Mattermost on Mattermost:
  - Change to `[Mattermost] Your password has been updated`
5. [Mattermost] test-email invited you to join pr5198 Team on Mattermost:
  - Change to `[Mattermost] test-email invited you to join pr5198 Team` (Useful to keep the team name there for this one, do you think?)
6. [Mattermost] Your MFA has been updated on Mattermost:
  - Change to `[Mattermost] Your MFA has been updated`

The subjects that don't appear to me to need updating:
1. [Mattermost] Reset your password
2. [Mattermost] You joined i-064b9b30bc2c0b9a1.spinmint.com:8065
3. [Mattermost] Email Verification
4. [Mattermost] New Direct Message in from linda-test2 on February 28, 2017 (Unless we think the date is superfluous?)
5. [Mattermost] New Mention in pr5198 (Town Square) on February 28, 2017 (Unless we think the date is superfluous?)

**Comment by jasonblais** [comment]
Thanks @lindalumitchell! Agree on the proposals

@cpanato I was preparing a pull request against your branch for the string changes, so you could review and approve/comment on them. However, I kept getting 404 errors for some reason when I tried

image

So instead, I had to push the changes. Sorry about that, hope that's okay? Let me know if the changes look good and I'll re-queue this for review with Corey.

**Comment by cpanato** [comment]
@jasonblais sorry about what?

**Comment by jasonblais** [comment]
@coreyhulen ready for review. @cpanato addressed your comments.

The only other updates since were a few string changes to the email subjects that remove redundant references to team and server names

**Pull request merged**